### DISCUSSION TITLE
Error: ENOENT: no such file or directory, lstat '/vercel/path0/node_modules\styled-jsx\index.js'

### DISCUSSION THREAD

**Discussion by Pawel-Gnat** [vercel/community#6421] [Help]
### Summary

I have a problem with building a project to Vercel using Github Actions. Locally, the project using Vercel CLI builds without any problem (the project without the help of GHA, by connecting github to vercel platform also builds). However, I get an error during automation. I've already added a notation to the env about VERCEL_FORCE_NO_BUILD_CACHE: 1, but that didn't help either, and I have no idea what to do next anymore.

```
Created all serverless functions in: 4.616s
Collected static files (public/, static/, .next/static): 9.354ms
Build Completed in .vercel\output [45s]
Vercel CLI 33.7.0
Queued
Building
Error: ENOENT: no such file or directory, lstat '/vercel/path0/node_modules\styled-jsx\index.js'
Error: Process completed with exit code 1.
```

### Example

https://github.com/Pawel-Gnat/Fintax

### Steps to Reproduce

```name: GitHub Actions Vercel Preview Deployment

on:
  push:
    branches:
      - development

jobs:
  Deploy-Preview:
    runs-on: windows-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v4

      - name: Configure node
        uses: ./.github/actions/setup-project

      - name: Format & Lint code
        uses: ./.github/actions/format-code

      - name: Deploy app to Vercel preview env
        uses: ./.github/actions/vercel-deployment
        with:
          mode: 'preview'
          vercel-token: ${{ secrets.VERCEL_TOKEN }}
        env:
          VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
          VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
          VERCEL_FORCE_NO_BUILD_CACHE: 1
          DATABASE_URL: ${{ secrets.DATABASE_URL }}
          NEXTAUTH_SECRET: ${{ secrets.NEXTAUTH_SECRET }}
          NEXTAUTH_URL: ${{ secrets.NEXTAUTH_URL }}
          NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME: ${{ secrets.NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME }}
```

That's my GHA workflow. Maybe there is something wrong with my script file? But project has been sent and built properly, because I see a message from a Vercel platform.

![image](https://github.com/vercel/community/assets/104066590/9d6907c5-805a-4629-91de-1e7f65b957dc)

**Comment by OriginalEXE** [comment]
Having the same problem with the same dependency
![image](https://github.com/vercel/community/assets/2056251/0b9205ff-7540-42fa-b892-829435c36717)

**Reply by OriginalEXE** [reply]
Turns out it was because of the different Vercel CLI versions being used for local building and deploying to Vercel. The solution was to hardcode the Vercel CLI version everywhere, including the action I was using.

**Comment by swarnava** [comment]
It looks like your CLI version is pinned. Are you able to replicate by removing `VERCEL_CLI_VERSION` Environment variable from your project?
